core/trino-main/src/main/java/io/trino/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -253,7 +258,11 @@ void updateTaskStatus(TaskStatus newValue)
             onFail.accept(new TrinoException(REMOTE_TASK_MISMATCH, format("%s (%s)", REMOTE_TASK_MISMATCH_ERROR, HostAddress.fromUri(getTaskStatus().getSelf()))));
         }
 
-        dynamicFiltersFetcher.updateDynamicFiltersVersionAndFetchIfNecessary(newValue.getDynamicFiltersVersion());
+        synchronized (this) {
+            if (running) {
```

Comment:
Can it happen in non-aborted query?

I think that's still wrong, because last DF update could be potentially received after `ContinuousTaskStatusFetcher#stop` was called. We need that last request for DFs to happen, otherwise we loose filtering opportunity.

Reply:
You mean that `running` can get `true` bacasue `stop()` is called as a result of listener due to `taskStatus.setIf` above?

Maybe we should just move
```
dynamicFiltersFetcher.updateDynamicFiltersVersionAndFetchIfNecessary(newValue.getDynamicFiltersVersion());
```

to the top of `updateTaskStatus`. WDYT?